### PR TITLE
NP-7569 AnalyticEvent Model Setup

### PR DESCRIPTION
**Pull request by xuerongNanopay**

### CODE REVIEW

**Review by sarthak-marwaha — APPROVED** [review]

File: src/foam/nanos/analytics/AnalyticEvent.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      class: 'String',
+      name: 'traceId'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.session.Session',
+      name: 'sessionId'
+    },
+    {
+      class: 'Object',
```
Comment (sarthak-marwaha) [inline src/foam/nanos/analytics/AnalyticEvent.js:43]: @jlhughes Should this be a reference?

File: src/foam/nanos/analytics/AnalyticEvent.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    {
+      class: 'String',
+      name: 'name'
+    },
+    { 
+      class: 'DateTime',     
+      name: 'timestamp',
+    },
+    {
+      class: 'Int',
```
Comment (jlhughes) [inline src/foam/nanos/analytics/AnalyticEvent.js]:
```suggestion
      class: 'Duration',
```

File: src/foam/nanos/analytics/AnalyticEvent.js (hunk 1 above)
Comment (kgrgreer) [reply src/foam/nanos/analytics/AnalyticEvent.js:43]: A reference to what?

**Comment by kgrgreer** [comment]
LGTM

File: src/foam/nanos/analytics/AnalyticEvent.js (hunk 1 above)
Comment (jlhughes) [reply src/foam/nanos/analytics/AnalyticEvent.js:43]: @sarthak-marwaha reference to what?  And we can't have one property be a Reference to multiple other types, so no, a reference would not work. 
This is the objects id, and if the id is generated with FUID then we can use FUID to figure out what DAO it came from.

Comment (sarthak-marwaha) [reply src/foam/nanos/analytics/AnalyticEvent.js:43]: Right....references need a single DAO, maybe FObjectProperty then, If it is just IDs then class should be String

**Review by sarthak-marwaha — APPROVED** [review]

**Review by jlhughes — APPROVED** [review]

**Pull request merged**